nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/test/java/org/apache/nifi/distributed/cache/server/map/TestMapCacheClient.java

```diff
@@ -0,0 +1,146 @@
+ * limitations under the License.
+ */
+package org.apache.nifi.distributed.cache.server.map;
+
+import org.apache.commons.lang3.SerializationException;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.distributed.cache.client.DistributedMapCacheClientService;
+import org.apache.nifi.distributed.cache.client.Serializer;
+import org.apache.nifi.distributed.cache.operations.MapOperation;
+import org.apache.nifi.distributed.cache.protocol.ProtocolVersion;
+import org.apache.nifi.distributed.cache.server.EvictionPolicy;
+import org.apache.nifi.distributed.cache.server.codec.CacheOperationResultEncoder;
+import org.apache.nifi.distributed.cache.server.codec.CacheVersionRequestHandler;
+import org.apache.nifi.distributed.cache.server.codec.CacheVersionResponseEncoder;
+import org.apache.nifi.distributed.cache.server.codec.MapCacheRequestDecoder;
+import org.apache.nifi.distributed.cache.server.codec.MapCacheRequestHandler;
+import org.apache.nifi.distributed.cache.server.codec.MapRemoveResponseEncoder;
+import org.apache.nifi.distributed.cache.server.codec.MapSizeResponseEncoder;
+import org.apache.nifi.distributed.cache.server.codec.MapValueResponseEncoder;
+import org.apache.nifi.event.transport.EventServer;
+import org.apache.nifi.event.transport.configuration.ShutdownQuietPeriod;
+import org.apache.nifi.event.transport.configuration.TransportProtocol;
+import org.apache.nifi.event.transport.netty.NettyEventServerFactory;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.processor.Processor;
+import org.apache.nifi.remote.StandardVersionNegotiator;
+import org.apache.nifi.remote.VersionNegotiator;
+import org.apache.nifi.remote.io.socket.NetworkUtils;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.MockConfigurationContext;
+import org.apache.nifi.util.MockControllerServiceInitializationContext;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestMapCacheClient {
+    private int port;
+    private EventServer server;
+    private final Serializer<String> stringSerializer = new StringSerializer();
+
+    @BeforeEach
+    public void setRunner() {
+        final TestRunner runner = TestRunners.newTestRunner(Mockito.mock(Processor.class));
+        port = NetworkUtils.getAvailableTcpPort();
+
+        final NettyEventServerFactory serverFactory = new NettyEventServerFactory(
+                InetAddress.getLoopbackAddress(), port, TransportProtocol.TCP);
```

Comment:
The `InetAddress.getLoopbackAddress()` call can behave differently in some environments. Instead, recommend using `InetAddress.getByName("127.0.0.1")` and also using `127.0.0.1` as the client address to avoid any name resolution concerns.

Reply:
Using `127.0.0.1` is the most stable across platforms, specifically Windows. Alternatively, we could keep this `InetAddress.getLoopbackAddress()`, and use the value from `getHostAddress()` in the client setup, instead of `localhost`.